gencred/README.md

```diff
@@ -143,11 +143,14 @@ users:
 If you store kubeconfig files in kubernetes secrets to allow pods to access other kubernetes clusters (like many of Prow's components require) consider using [`merge_kubeconfig_secret.py`](/gencred/merge_kubeconfig_secret.py) to merge the kubeconfig produced by `gencred` into the secret.
 
 ```shell
-# Generate a kubeconfig.yaml as described and shown above, then run something like:
-./merge_kubeconfig_secret.py --src-key=config-old --dest-key=config-new kubeconfig.yaml
-# Update references (e.g. `--kubeconfig` flags) to point to config-new instead of config-old.
+# Generate a kubeconfig.yaml as described and shown above.
+./merge_kubeconfig_secret.py --auto --context=my-kube-context kubeconfig.yaml
```

Comment:
To avoid the need for this, the `KUBECONFIG` env var can be used to supply mulitple kubeconfigs: https://github.com/openshift/release/blob/d534072d3f09e94d1a070469dd27a7da2b1d1b63/clusters/app.ci/prow/03_deployment/sinker.yaml#L36

Has the nice sideeffect that adding/removing a cluster causes a rollout and secret size is not a concern, as each kubeconfig can get its own secret

Reply:
That is something I considered originally and that is how the script actually does the merge, but I figured that a single kubeconfig would be easier to manage. Here are my thoughts, let me know what you think:
- I prefer using flags rather than using env vars since it is clear how flags are consumed.
- Order matters when using the `KUBECONFIG` env var and accidentally adding a kubeconfig to the front of the list instead of the back would cause lots of problems due to the `current-context` changing.

> Has the nice sideeffect that adding/removing a cluster causes a rollout and secret size is not a concern, as each kubeconfig can get its own secret

- Adding a cluster using `./merge_kubeconfig_secret.py` and updating the `--kubeconfig` flag causes a rollout in the same way that adding/updating a secret and updating the `KUBECONFIG` env var would.
- Having a separate secret for every kubeconfig would require a lot of extra mounts in the deployment files which isn't a problem, but would be more annoying to read than just a single kubeconfig secret mount.
- Secret size doesn't really seem like a concern for our use cases at least. The k8s-prow kubeconfig is only 36KB and the secret size limit is 1MB. If an instance does eventually reach the secret size limit we could start using the `KUBECONFIG` env var with multiple secrets, but even then I think we'd want to consolidate into a small number of kubeconfigs rather than a kubeconfig per cluster since we'd have a huge number of clusters if we are hitting the size limit.

Reply:
> Order matters when using the KUBECONFIG env var and accidentally adding a kubeconfig to the front of the list instead of the back would cause lots of problems due to the current-context changing.

I would have expected us to never care about `current-context?



Generally I prefer to have it explicitly and see in the config which clusters are referenced rather than having to look into the contents of a secret. But at the end of the day its probably a question of taste, both work.

/lgtm